ReactJS Exception Handler

awylie199 opened the pull request:
Hi,

At the moment rendering Exceptions are simply dumped to output by ReactJS  #35 .

I've added a very simple handler which simply throws the Exception that's raised so it's caught higher up (Whoops by default). I guess you could do something more clever here and let people pass in their own custom handler for closer control, but it's been ok for now.

![screen shot 2016-04-08 at 20 16 30](https://cloud.githubusercontent.com/assets/12545202/14395298/b522d806-fdc8-11e5-9d81-34f44874a4ff.png)

I've also added a small test to confirm that an exception is raised, for example when it can't find the component to render.

Thanks

awylie199 committed:
Added exception handler for ReactJS to throw any errors, instead of echoing them, to be caught by the default handler (e.g. Whoops). Added a test to confirm raised exception on a non-existent component.

awylie199 committed:
Forgot to add call to setErrorHandler

talyssonoc commented:
Very good! Thanks for your contribuition! I'm gonna make a small comment in case you can change that before the merge.

talyssonoc commented on tests/PropagatedExceptionTest.php:
```diff
@@ -0,0 +1,24 @@
+  }
+
+  /**
+   * @expectedException V8JsException
```
Could you use [explicit exception expectation](https://phpunit.de/manual/current/en/writing-tests-for-phpunit.html#writing-tests-for-phpunit.exceptions.examples.ExceptionTest.php) instead of use an annotation?

awylie199 committed:
Changed PropagatedExceptionTest to handle error explicitly rather than by notation. Upgraded phpunit dependency in composer to 5.3 for expectException test. Added tearDown to Exception test to close output buffers flagged as risky in new phpUnit.

awylie199 commented:
Hey,

No problem, I've changed the exception to the explicit version - but I had to upgrade the composer php unit dependency. The old dependency was at 4.0, and I don't think the new expectException method was added until 5.2.

I see this has caused the Travis tests to fail - do you think the php dependency should go up to 5.6?

I added the tearDown method on the phpUnit PropagatedExceptionTest because it was flagged as risky by phpunit 5.3:

<img width="770" alt="screen shot 2016-04-12 at 18 27 31" src="https://cloud.githubusercontent.com/assets/12545202/14469410/56ff54fe-00dc-11e6-9079-ea5567368a44.png">

I think when the error is raised it opens output buffering.

talyssonoc commented:
@awylie199 thanks for the contribution! I'm with some problems with my computer this week so I'll not be able to test it locally, but as soon as I get it back I'm gonna merge it.

awylie199 commented:
Hey, no problem, hope you manage to fix your computer ok!

jeroenverfallie commented:
@talyssonoc Have you had the chance to look at this yet?
